pkg/compact/compact.go

```diff
@@ -470,6 +483,177 @@ func (cg *Group) Resolution() int64 {
 	return cg.resolution
 }
 
+type ProgressMetrics struct {
+	NumberOfCompactionRuns   *prometheus.GaugeVec
+	NumberOfCompactionBlocks *prometheus.GaugeVec
+}
+
+type ProgressCalculator interface {
+	ProgressCalculate(ctx context.Context, groups []*Group) error
+}
+
+type CompactionSimulator struct {
```

Comment:
Let's be more consistent on the naming. If the interface is called `ProgressCalculator` then let's call it also `CompactionProgressCalculator` instead of `xxxSimulator` here.